Awful.apk/src/main/java/com/ferg/awfulapp/widget/PageBar.java

```diff
@@ -86,30 +86,54 @@ private void init() {
      * @param lastPage    the number of last page in the page range
      */
     public void updatePagePosition(int currentPage, int lastPage) {
-            /*
-                Possible states and their button layouts (left side / right side):
-                    single page -         refresh /
-                    first of many -       refresh / next
-                    middle of many - prev refresh / next
-                    last of many -   prev         / refresh2
-
-             */
-        String text = String.format(Locale.getDefault(), "Page %d%s", currentPage, lastPage >= FIRST_PAGE ? "/" + lastPage : "");
-//        String text = String.format(Locale.getDefault(), getResources().getString(R.string.page_bar_text), currentPage, lastPage >= FIRST_PAGE ? "/" + lastPage : "");
+        PageType type;
+        if (currentPage == FIRST_PAGE) {
+            type = (currentPage == lastPage) ? PageType.SINGLE : PageType.FIRST_OF_MANY;
+        } else if (currentPage == lastPage) {
+            type = PageType.LAST_OF_MANY;
+        } else {
+            type = PageType.ONE_OF_MANY;
+        }
+        // if currentPage is greater than lastPage, then lastPage isn't a meaningful page count (-1 is passed in when we don't have that data anyway)
+        boolean hasPageCount = lastPage >= currentPage;
+        updateDisplay(currentPage, lastPage, type, hasPageCount);
+    }
+
+    private void updateDisplay(int currentPage, int lastPage, PageType pageType, boolean hasPageCount) {
+        String text = String.format(Locale.getDefault(), "Page %d%s", currentPage, hasPageCount ? "/" + lastPage : "");
```

Comment:
I hope the irony of using string concatenation inside a String.format is not lost on you :V

Reply:
It's my boy ternary operator